### ISSUE TITLE
When file over 2GB bun doesn't throw an error but continues execution without file content.

### ISSUE THREAD

**Issue by arreehm** [oven-sh/bun#2305]
### What version of Bun is running?

0.5.6

### What platform is your computer?

Linux 6.2.0-2-MANJARO x86_64 unknown

### What steps can reproduce the bug?

1. Create file larger than 2GB, e.g. CSV
2. Read it using fileReadSync

Problem reproduced, file is like red, but not having content. No error thrown.

### What is the expected behavior?

Throwing an error of file too large, or reading file.

### What do you see instead?

No content of file/

**Comment by paperclover** [comment]
I tried to reproduce this but I cannot get any error or incorrect output when using `readFileSync`. Are you using buffers or strings? There might be issues with very big strings around the 2gb range if you're using strings, but I was able to run the following with no issue on a 4gb csv:

```ts
import fs from "fs";

const string = fs.readFileSync("big.csv", "utf-8");
console.log(string);
```

The buffer approach also seems to work.

Bun v0.5.8, `Linux 6.2.2-arch1-1 x86_64`
